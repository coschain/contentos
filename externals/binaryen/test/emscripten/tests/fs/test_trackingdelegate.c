#include <stdio.h>
#include <stdlib.h>
#include <emscripten.h>

int main() {

  EM_ASM(
    FS.trackingDelegate['willMovePath'] = function(oldpath, newpath) {
      Module.print('About to move "' + oldpath + '" to "' + newpath + '"');
    };
    FS.trackingDelegate['onMovePath'] = function(oldpath, newpath) {
      Module.print('Moved "' + oldpath + '" to "' + newpath + '"');
    };
    FS.trackingDelegate['willDeletePath'] = function(path) {
      Module.print('About to delete "' + path + '"');
    };
    FS.trackingDelegate['onDeletePath'] = function(path) {
      Module.print('Deleted "' + path + '"');
    };
    FS.trackingDelegate['onOpenFile'] = function(path, flags) { 
      Module.print('Opened "' + path + '" with flags ' + flags);
    };
    FS.trackingDelegate['onWriteToFile'] = function(path) {
      Module.print('Wrote to file "' + path + '"');
    };
  );

  FILE *file;
  file = fopen("/file.txt", "w");
  fputs("hello!", file);
  fclose(file);
  rename("/file.txt", "/renamed.txt");
  file = fopen("/renamed.txt", "r");
  char str[256] = {};
  fgets(str, 255, file);
  printf("File read returned '%s'\n", str);
  fclose(file);
  remove("/renamed.txt");
}
