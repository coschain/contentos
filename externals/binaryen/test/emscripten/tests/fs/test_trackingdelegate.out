Opened "/file.txt" with flags 2
Wrote to file "/file.txt"
Wrote to file "/file.txt"
About to move "/file.txt" to "/renamed.txt"
Moved "/file.txt" to "/renamed.txt"
Opened "/renamed.txt" with flags 1
Wrote to file "/dev/tty"
File read returned 'hello!'
Wrote to file "/dev/tty"
About to delete "/renamed.txt"
Deleted "/renamed.txt"
